acceptance/Rakefile

```diff
@@ -85,88 +96,40 @@ def beaker_test(mode = :packages, options = {})
     mv(options_file, "log/latest")
     puts "\n\n"
   end
 end
 
-def generate_config_for_latest_hosts
-  preserved_config_hash = { 'HOSTS' => {} }
-
-  config_hash = YAML.load_file('log/latest/config.yml').to_hash
-  nodes = config_hash['HOSTS'].map do |node_label,hash|
-    { :node_label => node_label, :platform => hash['platform'] }
-  end
-
-  pre_suite_log = File.read('log/latest/pre_suite-run.log')
-  nodes.each do |node_info|
-    possible_hostname_match = /^(\w+) \(#{node_info[:node_label]}\)/.match(pre_suite_log)
-    hostname = (possible_hostname_match || Array.new)[1]
-    fqdn = hostname ? "#{hostname}.delivery.puppetlabs.net" : "unknown"
-    preserved_config_hash['HOSTS'][fqdn] = {
-      'roles' => [ 'agent'],
-      'platform' => node_info[:platform],
-    }
-    preserved_config_hash['HOSTS'][fqdn]['roles'].unshift('master') if node_info[:node_label] =~ /master/
-  end
-
-  puts "\n\n"
-  pp preserved_config_hash
-  puts "\n\n"
-
-  File.open('log/latest/preserved_config.yaml', 'w') do |config_file|
-    YAML.dump(preserved_config_hash, config_file)
-  end
-rescue Errno::ENOENT => e
-  puts "Couldn't generate log #{e}"
-end
-
-def list_preserved_configurations(secs_ago = ONE_DAY_IN_SECS)
-  preserved = {}
-  Dir.glob('log/*_*').each do |dir|
-    preserved_config_path = "#{dir}/preserved_config.yaml"
-    yesterday = Time.now - secs_ago.to_i
-    if preserved_config = File.exists?(preserved_config_path)
-      directory = File.new(dir)
-      if directory.ctime > yesterday
-        hosts = []
-        preserved_config = YAML.load_file(preserved_config_path).to_hash
-        preserved_config['HOSTS'].each do |hostname,values|
-          hosts << "#{hostname}: #{values['platform']}, #{values['roles']}"
-        end
-        preserved[hosts] = directory.to_path
-      end
-    end
-  end
-  preserved.map { |k,v| [v,k] }.sort { |a,b| a[0] <=> b[0] }.reverse
-end
-
 def list_preserved_hosts(secs_ago = ONE_DAY_IN_SECS)
   hosts = Set.new
-  Dir.glob('log/**/pre*suite*run.log').each do |log|
-    yesterday = Time.now - secs_ago.to_i
-    File.open(log, 'r') do |file|
-      if file.ctime > yesterday
-        file.each_line do |line|
-          matchdata = /^(\w+) \(.*?\) \$/.match(line.encode!('UTF-8', 'UTF-8', :invalid => :replace))
-          hosts.add(matchdata[1]) if matchdata
+  Dir.glob('log/*').each do |dir|
+    unless dir =~ /^log\/latest$/ 
+      Dir.glob("#{dir}/*").each do |date_dir|
+        preserved_config_path = "#{date_dir}/#{SUTS_FILENAME}"
+        if preserved_config = File.exists?(preserved_config_path)
+          directory = File.new(date_dir)
+          yesterday = Time.now - secs_ago.to_i
+          if directory.ctime > yesterday
+            File.open(preserved_config_path, 'r') do |file|
+              if file.ctime > yesterday
+                file.each_line do |line|
+                  matchdata = /(\w+\.delivery\.puppetlabs\.net) \(.*?\)$/.match(line.encode!('UTF-8', 'UTF-8', :invalid => :replace))
```

Comment:
inherited... i forget if this even works.  
i can certainly remove the `encode`.

most of this functionality can/should move to beaker/vmpooler API.
which leads me to investigating a rakefile gem for at least puppet/facter/hiera, and i'd love to jettison the acceptance Rakefile altogether as in other projects.... but i digress